### ISSUE TITLE
Make a tutorial notebook for a spectral analysis with obs grouping

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1577]
This is a reminder issue to follow up on #1575 .

I think we should change this docs page:

https://github.com/cdeil/gammapy/blob/75b9321d3f385885483e44ef49278ac78545cad8/docs/data/obs_group.rst

to a tutorial notebook, and actually show a working example of a spectrum analysis with runs grouped, as hinted at in the lower part of that page.

Extra points if updating an existing notebook like
http://docs.gammapy.org/dev/notebooks/spectrum_pipe.html
or
http://docs.gammapy.org/dev/notebooks/spectrum_analysis.html
instead of just adding a new one.

Update to use real HESS runs can be done at the same time, next week I'll put the files from
https://github.com/cdeil/HESS-DL3-DR1-preview in gammapy-extra and release.

If someone does this next week for v0.8, great!
Otherwise I'll try to do it.
And if there's no time, we'll defer to v0.9, this is not a blocker.

cc @leajouvin @registerrier

**Comment by cdeil** [comment]
Just now I found an old example running a spectral analysis with observation grouping in `examples/test_group_observations.py`. I removed that in ff1b00ff387ade56ccf49638e2a5a3499198b5f1, like I said above, we should add a tutorial notebook for that.

**Comment by cdeil** [comment]
I'm cleaning up the RST docs a bit first before tending to the notebooks.
Removed `docs/data/obs_group.rst` in 404ac495
That should be the starting point for this tutorial notebook on obs grouping.

**Comment by cdeil** [comment]
@registerrier - Do you have something here for Monday or Tuesday latest? IMO it's not critical, OK to just postpone to v0.9.

**Comment by registerrier** [comment]
There are two different cases here:
- grouping observations to perform data reduction within specific range of parameters. 
- grouping datasets after data reduction for specific analysis purposes

Here I think the main case is the second. Grouping datasets is useful in different occasions. It is used for:
- light curve to group datasets in time bins
- spectral analysis to group data
For this to be possible it is necessary to keep track of the information on which to do the grouping. This is still lacking. There was an attempt to add something here #2447 . We have to find a good solution for this to solve this issue.

**Comment by adonath** [comment]
@QRemy worked on a hierarchical clustering method for run grouping in #4242. Once the PR is merged we could show the clustering in our docs...will assign this issue to you @QRemy and re-label to maybe v1.1.

**Comment by adonath** [comment]
Just for completeness I think we could do:
- Show simple grouping based on zenith angle
- Show hierarchical clustering introduced in #4242 
- Show some other method of choice from sklearn, could be something simple like kmeans as well...
- Viusalizing the feature space would be nice to see how the observations really cluster in e.g. a 2D space and then color code with zenith angle...not sure whether we have enough public data for this...

**Comment by registerrier** [comment]
See also issue #4788 to expose existing feature such as observation clustering

**Comment by Astro-Kirsty** [comment]
This should be covered now by https://github.com/gammapy/gammapy/pull/4810
